ran-core/src/main/java/io/ran/Clazz.java

```diff
@@ -466,15 +464,28 @@ public ClazzMethodList methods() {
 					result.put(m, new ClazzMethod(this, m));
 				});
 			} else {
+				Clazz finalWorking = working;
 				Arrays.stream(working.clazz.getDeclaredMethods()).filter(m -> !m.isBridge()).forEach(m -> {
-					result.put(m, new ClazzMethod(this, m));
+					result.put(m, new ClazzMethod(this, finalWorking, m));
+				});
+				Arrays.stream(working.clazz.getInterfaces())
+						.flatMap(inter -> Stream.of(inter.getMethods()))
+						.filter(m -> !m.isBridge())
+						.filter(m -> result.keySet().stream().noneMatch(otherM -> getSignatureOfMethod(m).equals(getSignatureOfMethod(otherM))))
+						.forEach(m -> {
+					result.put(m, new ClazzMethod(this, finalWorking, m));
```

Comment:
@anton-larsen-persequor this feature was missing, interfaces may have default implementations - which were not returned in the list of methods of a Clazz. The reason for filtering based on Signature is that the AutoMapper cannot deal with the same method signature showing up more than once (even though the Clazz arguably may have multiple implementations, the default one and the overwritten one).